### PR TITLE
Migration to Prometheus Aggregator

### PR DESCRIPTION
**Pull request by balbirthomas**
This pull request Migrates the LMA Proxy operator to use the recently added `PrometheusAggregator` object.

### CODE REVIEW

**Comment by balbirthomas** [comment]
Please note `tox` based tests are actually passing. The test failure show is a red herring because one of the commits in this pull request removes the `run_tests` script and main branch uses it to run tests.

**Review by mmanciop — APPROVED** [review]

File: .flake8 [hunk 1]
```diff
@@ -1,6 +1,8 @@
 [flake8]
```
Comment (jnsgruk) [inline .flake8:1]: This file should be dropped in favour of the generic `pyproject.toml` config, no?

File: tox.ini [hunk 2]
```diff
@@ -12,10 +12,11 @@ exclude =
     build,
```
Comment (jnsgruk) [inline tox.ini:12]: Please migrate this to the new standard we have set, even if in a seperate PR :)

File: .flake8 (hunk 1 above)
Comment (jnsgruk) [reply .flake8:1]: Can be in a seperate PR with tox config :)

**Review by jnsgruk — APPROVED** [review]

File: .flake8 (hunk 1 above)
Comment (balbirthomas) [reply .flake8:1]: Sure, I will create another PR with pyproject.toml etc. I need to spend some time understanding all that. This one is just to get the core functionality in. I am still waiting on the Prometheus library rename fix after which I need another PR.

File: tox.ini (hunk 2 above)
Comment (balbirthomas) [reply tox.ini:12]: Yes will do.

File: .flake8 [hunk 3]
```diff
@@ -1,6 +1,8 @@
...
 max-line-length = 99
-select: E,W,F,C,N
+max-complexity = 10
+select = E,W,F,C,N
```
Comment (sed-i) [inline .flake8:4]: This selection will not warn for docstrings and other things. Check out same file in alertmanager/karma.

Comment (balbirthomas) [reply .flake8:4]: Thanks. Will fix in a future commit.

**Pull request merged**